cli/azd/pkg/tools/docker/docker.go

```diff
@@ -90,11 +93,28 @@ func (d *docker) Build(
 
 	args = append(args, buildContext)
 
-	res, err := d.executeCommand(ctx, cwd, args...)
+	// Build and produce output
+	runArgs := exec.NewRunArgs("docker", args...).WithCwd(cwd)
+	d.console.StopSpinner(ctx, "", input.Step)
+	previewer := d.console.ShowPreviewer(ctx,
+		&input.ShowPreviewerOptions{
+			Prefix:       "    ",
+			MaxLineCount: 8,
+			Title:        "Docker Output",
+		})
+	runArgs.StdOut = previewer
+
+	_, err := d.commandRunner.Run(ctx, runArgs)
+	previewer.Stop(ctx)
 	if err != nil {
 		return "", fmt.Errorf("building image: %w", err)
 	}
 
+	// Build should just return cached img id
+	res, err := d.executeCommand(ctx, cwd, append(args, "-q")...)
+	if err != nil {
+		return "", fmt.Errorf("building image: %w", err)
+	}
```

Comment:
That's what it looks like to me as well.  We should investigate whether there is a way we can ensure we only build the image one time but still get the return values that we need.

Reply:
I tried parsing the docker-build output to get the same image-id that is returned by docker-build when using `-q`, but it becomes nasty, ugly and fragile (as there's not a contract from docker-build output. It can be changed at any moment).  On the other hand, when using `-q`, there is a contract from docker-cli to return only the img-id.

Note that, even though we call docker-build twice, the second time takes less than a second to complete, as it basically just returns the image-id.

Reply:
I agree that with caching this operation would be pretty fast.

It looks like `docker build` supports a flag to write the image id to a file after it completes.

From `docker build --help`:

```
      --iidfile string                Write the image ID to the file
```

We could allocate a temp file, and then pass the path to it to docker build and then read the result once the command completes.

Docker 17.06.0, so we should be able to take a dependency on it.

Reply:
If possible, I would like not to depend on writing a file, as it would require to check for a possible error during the file write/read.

Would it be really better to depend on the file, than calling `build` twice?  (knowing that the second call to `build` is basically a no-op call just to get the value we need)

Reply:
> If possible, I would like not to depend on writing a file, as it would require to check for a possible error during the file write/read.
> 
> Would it be really better to depend on the file, than calling build twice? (knowing that the second call to build is basically a no-op call just to get the value we need)

I think I have a slight preference here.  Doing a single docker build and fetching the id from that operation feels like it is safer than running the operation twice and hoping that it is fast due to caching.